server/src/main/java/org/elasticsearch/cluster/metadata/ProjectMetadata.java

```diff
@@ -2199,9 +2167,11 @@ public static ProjectMetadata readFrom(StreamInput in) throws IOException {
 
         readProjectCustoms(in, builder);
 
-        int reservedStateSize = in.readVInt();
-        for (int i = 0; i < reservedStateSize; i++) {
-            builder.put(ReservedStateMetadata.readFrom(in));
+        if (in.getTransportVersion().before(TransportVersions.PROJECT_RESERVED_STATE_MOVE_TO_REGISTRY)) {
+            int reservedStateSize = in.readVInt();
+            for (int i = 0; i < reservedStateSize; i++) {
+                ReservedStateMetadata.readFrom(in);
+            }
         }
```

Comment:
This simply drops the project level `ReservedStateMetadata` from an old node. Did we explicilty decide to not support such a migration case? I guess that is OK since all MP clusters are temporary. Just want to confirm whether it is indeed intentional.